util/tracing/forwarder/forwarder.go

```diff
@@ -0,0 +1,28 @@
+package forwarder
+
+import (
+	"context"
+
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+)
+
+type Exporter struct {
+	bsp sdktrace.SpanProcessor
+}
+
+func NewExporter(exp sdktrace.SpanExporter) *Exporter {
+	return &Exporter{
+		bsp: sdktrace.NewBatchSpanProcessor(exp),
```

Comment:
question: This seems to drop spans if the processor's queue becomes full. Should we initialize it to block instead in that scenario?

Reply:
It seems the buffer size is [2048 spans](https://github.com/fiam/buildkit/blob/c0205ad636507d0f98a9dde4e08215212ea24689/vendor/go.opentelemetry.io/otel/sdk/trace/batch_span_processor.go#L20), while buildx creates [~1 span per target](https://github.com/search?q=repo%3Adocker%2Fbuildx%20StartSpan&type=code). We should be safe even if exporting the spans upstream from BuildKit is a bit slow.

Reply:
I also don't think slow is an issue with the processor. I think it's mostly if the export fails where it becomes an issue. If the connection is slow I think it should just wait.